### PR TITLE
security group: make listable

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Signed-off-by: Pierre-Emmanuel Jacquier <pierre-emmanuel.jacquier@epitech.eu>

### CODE REVIEW

File: security_groups.go [hunk 1]
```diff
@@ -283,8 +144,42 @@ func (*ListSecurityGroups) response() interface{} {
...
+func (sg *SecurityGroup) ListRequest() (ListCommand, error) {
+	//TODO add tags
+	//TODO look to implem VirtualMachineID
+	req := &ListSecurityGroups{
+		Account:           sg.Account,
+		DomainID:          sg.DomainID,
+		ID:                sg.ID,
+		ProjectID:         sg.ProjectID,
+		SecurityGroupName: sg.Name,
+		//VirtualMachineID: sg.VirtualMachineIDs
```
Comment (greut) [inline security_groups.go]: that's not part of the our API.

```
$ cs listApis name=listSecurityGroups
```

Comment (pierre-emmanuelJ) [reply security_groups.go]: So i remove it ?

Comment (greut) [reply security_groups.go]: It won't exist in the near future

Comment (greut) [reply security_groups.go]: But we have an issue here.

File: security_groups.go [hunk 2]
```diff
@@ -283,8 +144,40 @@ func (*ListSecurityGroups) response() interface{} {
 	return new(ListSecurityGroupsResponse)
 }
 
-// ListSecurityGroupsResponse represents a list of security groups
-type ListSecurityGroupsResponse struct {
-	Count         int             `json:"count"`
-	SecurityGroup []SecurityGroup `json:"securitygroup"`
+// ListRequest builds the ListSecurityGroups request
+func (sg *SecurityGroup) ListRequest() (ListCommand, error) {
```
Comment (greut) [inline security_groups.go]: Move it next to `Get` and refactor `Get` to use`ListRequest`.

File: security_groups.go [hunk 3]
```diff
@@ -19,28 +19,44 @@ func (sg *SecurityGroup) Get(ctx context.Context, client *Client) error {
...
-	} else if count > 1 {
+	} else if listSize > 1 {
 		return fmt.Errorf("More than one SecurityGroup was found. Query: id: %s, name: %s", sg.ID, sg.Name)
 	}
 
-	return copier.Copy(sg, sgs.SecurityGroup[0])
+	securGroup, ok := resp[0].(*SecurityGroup)
+	if !ok {
+		return fmt.Errorf("SecurityGroup expected, got %t", resp[0])
+	}
```
Comment (greut) [inline security_groups.go]: good initiative :+1:

**Pull request merged**